### PR TITLE
# If applied, this commit will make life better for everyone

### PR DESCRIPTION
**Pull request by freddyC**
### Why is this change needed?
To help with peace on earth

### How does it address the issue?
takes test from 90 seconds to 4 seconds

A GIF or MEME to give some spice of the internet

## Description

I remove tests that use the `trainAsync` method.
I add tests that verify that `train` and `trainAsync` in the `NeuralNetwork` class both use the same internal methods.

## Motivation and Context

[issue](https://github.com/BrainJS/brain.js/issues/154)

## How Has This Been Tested?

`yarn test` seems to do the job.

## Types of changes

- [ ] ~Bug fix (non-breaking change which fixes an issue)~
- [x] New feature (non-breaking change which adds functionality)
- [ ] ~Breaking change (fix or feature that would cause existing functionality to not work as expected)~

## Author's Checklist:

- [x] My code focuses on the main motivation and avoids scope creep.
- [x] My code passes current tests and adds new tests where possible.
- [x] My code is [SOLID](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)) and [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).
- [x] I have updated the documentation as needed.

## Reviewer's Checklist:
- [ ] I kept my comments to the author positive, specific, and productive.
- [ ] I tested the code and didn't find any new problems.
- [ ] I think the motivation is good for the project.
- [ ] I think the code works to satisfies the motivation.

### CODE REVIEW

File: test/base/trainopts.js [hunk 1]
```diff
@@ -74,102 +74,78 @@ describe('train() options', () => {
   });
 });
 
+describe('train() and trainAsync() use the same internal methods', () => {
+  
+  it(' _prepTraining()', (done) => {
+    let trainingData = [ { input: [0, 0], output: [0] } ];
+    var net = new brain.NeuralNetwork();
+    let tick = 0;
+    net._prepTraining = (d, o) => {
```
Comment (robertleeplummerjr) [inline test/base/trainopts.js]: Why is this calling itself?  It stack is going to get pretty deep, pretty fast.

Comment (freddyC) [reply test/base/trainopts.js]: The tests pass, don't ask!  All we care about are the 
image marks.

Actually I think I am doing it wrong.

Comment (robertleeplummerjr) [reply test/base/trainopts.js]: lol

After talking privately, I believe sinon will be your guid here.  I look forward to seeing this mature.

File: test/base/trainopts.js [hunk 2]
```diff
@@ -74,102 +75,108 @@ describe('train() options', () => {
...
+  function getSpy (net, func) {
+    sinon.spy(net, func);
+  }
+
+  function killSpy (net, func) {
+    net[func].restore();
+  }
+
+  it('_prepTraining()', (done) => {
+    getSpy(net, '_prepTraining');
```
Comment (robertleeplummerjr) [inline test/base/trainopts.js]: Can you move setup and teardown to `before(() => {});` or `after(() => {});`?

File: test/base/trainopts.js [hunk 3]
```diff
@@ -74,102 +75,108 @@ describe('train() options', () => {
...
-    let p1 = net
-      .trainAsync(data, { errorThresh: 0.2 })
-      .then (res => {
-        assert.ok(res.error < 0.2, `[res.error, ${res.error}] should have been less then 0.2`);
+describe('train() and trainAsync() use the same private methods', () => {
+  let trainingData = [{ input: [0, 0], output: [0] }];
+  let opts = { iterations:1 };
+  let net = new brain.NeuralNetwork();
+
+  function getSpy (net, func) {
```
Comment (robertleeplummerjr) [inline test/base/trainopts.js]: because of how simple it is, can you just call `sinon.spy` and `.restore()` directly?

Comment (freddyC) [reply test/base/trainopts.js]: but what if I want to kill the spy?

**Review by robertleeplummerjr — APPROVED** [review]
good stuff here!

**Pull request merged**